compiler_opt/rl/compilation_runner.py

```diff
@@ -355,14 +333,12 @@ def collect_data(
       compilation_runner.ProcessKilledException is passed through.
       ValueError if example under default policy and ml policy does not match.
     """
-    cancellation_manager = self._get_cancellation_manager(cancellation_token)
-
     if reward_stat is None:
       default_result = self._compile_fn(
```

Comment:
Might want to add a lock to protect calls to `compile_fn`/`start_cancellable_process` so we don't wind up spawning a bunch of clang processes.

Like if a worker had `collect_data` called on it 100 times, those 100 calls could all try to happen simultaneously since ThreadPoolExecutor (in `local_worker_manager`) has no `max_workers` (even if it did, not sure how that'd work when threads are sleeping)

Also if a lock were added, it kinda throws a wrench into any attempted ordering of the input jobs (e.g. sorting by size in hopes its proportional to compile time), this is more of a problem for when implementing load balancing though

Reply:
but I want the 100 calls simultaneously.